week_11/case_study_11.Rmd

```diff
@@ -1,7 +1,72 @@
 ---
 title: "Case Study 11"
-author: Your Name
-date: August 1, 2020
+author: Ting Chang
+date: November 17, 2020
 output: github_document
 ---
- 
\ No newline at end of file
+
+# Racial Distribution in Buffalo, NY
+
+### Set Up
+
+```{r set-up, message=FALSE, warning=FALSE}
+library(tidyverse)
+library(spData)
+library(sf)
+library(mapview) # new package that makes easy leaflet maps
+library(foreach)
+library(doParallel)
+library(tidycensus)
+
+registerDoParallel(4)
+getDoParWorkers() # check registered cores
+```
+
+### Download Census Data
+```{r census-data, message=FALSE, warning=FALSE, results='hide'}
+racevars <- c(White = "P005003", 
+              Black = "P005004", 
+              Asian = "P005006", 
+              Hispanic = "P004003")
+
+options(tigris_use_cache = TRUE)
+erie <- get_decennial(geography = "block", variables = racevars, 
+                  state = "NY", county = "Erie County", geometry = TRUE,
+                  summary_var = "P001001", cache_table=T)
+```
+
+### Crop the Downloaded Data
+```{r crop, message=FALSE, warning=FALSE}
+boundary <- c(xmin=-78.9,xmax=-78.85,ymin=42.888,ymax=42.92)
+erie_crop <- st_crop(erie, boundary)
+```
+
+
+### Generate Dots to Represent Racial Distribution
+```{r dots, message=FALSE, warning=FALSE}
+erie_crop$variable <- as.factor(erie_crop$variable)
+
+dot_output <- foreach(i=1:4, .combine='rbind', .packages = c("tidyverse","sf")) %dopar% {
+  
+  race <- levels(erie_crop$variable)[i]
+  
+  erie_crop %>%
+    filter(variable == race) %>%
+    st_sample(size=.$value) %>%
+    st_as_sf() %>%
+    mutate(variable=race)
+}
+
+# %dopar% keep running in several errors
+# solving by specifying package name
+# refers to:
+# "https://stackoverflow.com/questions/34022083/parallel-computation-loading-packages-in-each-thread-only-once/34030445#34030445"
+# "https://stackoverflow.com/questions/17157211/error-when-using-dopar-instead-of-do-in-r-package-doparallel"
+```
```

Comment:
great idea to leave notes like this when you solve a problem. This makes it easier to figure out next time...